src/test/scala/chiselTests/Direction.scala

```diff
@@ -35,4 +40,8 @@ class DirectionSpec extends ChiselPropSpec with ShouldMatchers {
      elaborate(new BadDirection)
     }
   }
+
+  property("Chisel2 directions should be deprecated") {
+    elaborate(new DeprecatedDirection)
```

Comment:
What is this testing? That the deprecation system doesn't just crash? Since it's not actually checking anything here, you should either document what this is doing, or perhaps remove it completely.
Though it may be possible to get a list of deprecation errors from elaborate (but that may require more plumbing?)